### PR TITLE
Add support to Rails 7.1

### PR DESCRIPTION
**Pull request by aovertus**
Update `activerecord` requirements to support Rails 7.1

### CODE REVIEW

**Review by sampatbadhe — APPROVED** [review]

**Comment by courtsimas** [comment]
@sampatbadhe @mbleigh what's the status on getting this merged?
Thanks in advance!

**Comment by tvongaza** [comment]
We've been running ActsAsTaggableOn with edge rails (updated to latest commit every week) for months without issue, now that edge rails is tagged at 7.1 our bundle install is breaking. Be great to have this merged in.

Follow up question, is the less than runtime dependency really required? Instead only checking the greater & equal to 6.0 condition? At very least consider changing it to something a bit more inclusive like '<= 7'.

File: acts-as-taggable-on.gemspec [hunk 1]
```diff
@@ -22,7 +22,7 @@ Gem::Specification.new do |gem|
     gem.post_install_message = File.read('UPGRADING.md')
   end
 
-  gem.add_runtime_dependency 'activerecord', '>= 6.0', '< 7.1'
+  gem.add_runtime_dependency 'activerecord', '>= 6.0', '< 7.2'
```
[the remark this answers is not in the document]
Comment (aovertus) [reply acts-as-taggable-on.gemspec:25]: I've check it yet but keep in mind that acts_as_taggable rely on ActiveRecord not Rails.

If I remember properly acts as tenant provide extension to more than just ActiveRecord and therefor requires Rails as a dependency not just one module

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,10 +10,11 @@ Each change should fall into categories that would affect whether the release is
 
 As such, _Breaking Changes_ are major. _Features_ would map to either major or minor. _Fixes_, _Performance_, and _Misc_ are either minor or patch, the difference being kind of fuzzy for the purposes of history. Adding _Documentation_ (including tests) would be patch level.
 
-### [v9.0.2) / unreleased](https://github.com/mbleigh/acts-as-taggable-on/compare/v9.0.1...master)
+### [v9.1.0) / unreleased](https://github.com/mbleigh/acts-as-taggable-on/compare/v9.0.1...master)
 * Features
   * [@glampr Add support for prefix and suffix searches alongside previously supported containment (wildcard) searches](https://github.com/mbleigh/acts-as-taggable-on/pull/1082)
   * [@donquxiote Add support for horizontally sharded databases](https://github.com/mbleigh/acts-as-taggable-on/pull/1079)
+  * [aovertus Remove restriction around ActiveRecord 7.x versions allowing support until next major is released](https://github.com/mbleigh/acts-as-taggable-on/pull/1110)
```
Comment (aovertus) [inline CHANGELOG.md:17]: @tvongaza does this make sense ? I think we should not allow ActiveRecord 8 as we don't know which breaking changes will comes with it ?

Comment (paulhklam1122) [reply CHANGELOG.md:17]: Agreed, can do `< 8` for now. Let's get this merged if possible please.

Comment (tvongaza) [reply CHANGELOG.md:17]: Looks good!

**Review by tvongaza — APPROVED** [review]

**Comment by tvongaza** [comment]
@seuros Any chance of doing a minor gem release to allow this gem to work with rails 7.1 & beyond?

**Comment by n-rodriguez** [comment]
Hi there! any news?

**Comment by seuros** [comment]
OK i will now

**Comment by n-rodriguez** [comment]
Thank you!

**Pull request merged**